packages/toolkit/src/http/httpClient.ts

```diff
@@ -61,8 +61,16 @@ export function createHttpClient(opts: HttpClientOptions = {}): AxiosInstance {
     retryDelay: (attempt, err) => {
```

Comment:
we will need to cap the resolved value to avoid the bot waiting too long for the next retry. or alternatively track the cumulative time taken to process all retries and throw upon exceeding that.